### PR TITLE
Simplify enabled checks for special domains

### PR DESCRIPTION
**Pull request by jonathanKingston**
**Reviewer:**

## Description:

Consistently disables all features on special domain names.

## Automated tests:
- [ ] Unit tests
- [ ] Integration tests

###### Reviewer Checklist:
- [ ] **Ensure the PR solves the problem**
- [ ] **Review every line of code**
- [ ] **Ensure the PR does no harm by testing the changes thoroughly**
- [ ] **Get help if you're uncomfortable with any of the above!**
- [ ] Determine if there are any quick wins that improve the implementation

###### PR Author Checklist:
- [ ] Get advice or leverage existing code
- [ ] Agree on technical approach with reviewer (if the changes are nuanced)
- [ ] Ensure that there is a testing strategy (and documented non-automated tests)
- [ ] Ensure there is a documented monitoring strategy (if necessary)
- [ ] Consider systems implications 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202876284063657

### CODE REVIEW

**Review by dharb — COMMENTED** [review]
Looks good, thanks for consolidating these checks and making them more consistent. One question about CTL content script injection

File: shared/js/background/events.es6.js [hunk 1]
```diff
@@ -422,7 +422,7 @@ if (manifestVersion === 2) {
 browser.webNavigation.onCommitted.addListener(details => {
     const tab = tabManager.get({ tabId: details.tabId })
 
-    if (!tab || tab.site.specialDomainName) {
+    if (!tab) {
```
[the remark this answers is not in the document]
Comment (jonathanKingston) [reply shared/js/background/events.es6.js:425]: Yeah this is within utils.getClickToPlaySupport(tab) within 435; actually we can remove the isBroken check also.

**Comment by jonathanKingston** [comment]
As mentioned on asana I'm going to wait until we have https://github.com/duckduckgo/duckduckgo-privacy-extension/pull/1363 that way we have a shared object for managing the tab loading status that can easily be accessed from Site.

We need to account for tab loading when checking for if we should apply the feature; if the tab has settled then we shouldn't apply protections to new tab pages.

File: shared/js/background/tab-manager.es6.js [hunk 2]
```diff
@@ -198,6 +199,8 @@ class TabManager {
         const tab = tabManager.get({ tabId: request.tabId })
 
         if (tab) {
+            tab.status = request.status
+            tab.site.status = tab.status
```
Comment (jonathanKingston) [inline shared/js/background/tab-manager.es6.js:203]: I think this tab.site.status can go in the bin now we have tabState to track it all :)

**Review by Bankrollshow — APPROVED** [review]

**Pull request closed without merging**